flink/v1.20/flink/src/main/java/org/apache/iceberg/flink/sink/shuffle/SortKeySerializer.java

```diff
@@ -52,8 +52,32 @@ class SortKeySerializer extends TypeSerializer<SortKey> {
   private final int size;
   private final Types.NestedField[] transformedFields;
 
+  private int version = SortKeySerializerSnapshot.CURRENT_VERSION;
+
   private transient SortKey sortKey;
 
+  SortKeySerializer(Schema schema, SortOrder sortOrder, int version) {
+    this.version = version;
+    this.schema = schema;
+    this.sortOrder = sortOrder;
+    this.size = sortOrder.fields().size();
+
+    this.transformedFields = new Types.NestedField[size];
+    for (int i = 0; i < size; ++i) {
+      SortField sortField = sortOrder.fields().get(i);
+      Types.NestedField sourceField = schema.findField(sortField.sourceId());
+      Type resultType = sortField.transform().getResultType(sourceField.type());
+      Types.NestedField transformedField =
+          Types.NestedField.of(
+              sourceField.fieldId(),
+              sourceField.isOptional(),
+              sourceField.name(),
+              resultType,
+              sourceField.doc());
+      transformedFields[i] = transformedField;
+    }
```

Comment:
This code part seems to be a duplicate for me?
What is the reason for not using the same underlying method for the 2 constructors?

Reply:
@pvary Thank you very much for your suggestion. Since all member variables are final, they can only be initialized through the constructor. Most of the initialization for SortKeySerializer uses the latest version, and only when restoring the serializer do we need to pass in the version number during initialization. Therefore, I added a new constructor. 

Another option is to remove the newly added constructor and use a setVersion method for injection. Because the version number is only needed for initialization in one place.

Which approach do you think is more appropriate?

Reply:
Could this work?
```
  SortKeySerializer(Schema schema, SortOrder sortOrder) {
    this(schema, sortOrder, SortKeySerializerSnapshot.CURRENT_VERSION);
  }
```

Reply:
@pvary Yes, it's feasible. I've made the adjustments. Could you please take another look to see if there are any other issues? Thanks ！